yolox/core/launch.py

```diff
@@ -59,128 +57,39 @@ def launch(
     """
     world_size = num_machines * num_gpus_per_machine
     if world_size > 1:
-        if int(os.environ.get("WORLD_SIZE", "1")) > 1:
-            dist_url = "{}:{}".format(
-                os.environ.get("MASTER_ADDR", None),
-                os.environ.get("MASTER_PORT", "None"),
-            )
-            local_rank = int(os.environ.get("LOCAL_RANK", "0"))
-            world_size = int(os.environ.get("WORLD_SIZE", "1"))
-            _distributed_worker(
-                local_rank,
-                main_func,
-                world_size,
-                num_gpus_per_machine,
-                num_machines,
-                machine_rank,
-                backend,
-                dist_url,
-                args,
-            )
-            exit()
-        launch_by_subprocess(
-            sys.argv,
-            world_size,
-            num_machines,
-            machine_rank,
-            num_gpus_per_machine,
-            dist_url,
-            args,
-        )
-    else:
-        main_func(*args)
-
+        # https://github.com/pytorch/pytorch/pull/14391
+        # TODO prctl in spawned processes
 
-def launch_by_subprocess(
-    raw_argv,
-    world_size,
-    num_machines,
-    machine_rank,
-    num_gpus_per_machine,
-    dist_url,
-    args,
-):
-    assert (
-        world_size > 1
-    ), "subprocess mode doesn't support single GPU, use spawn mode instead"
-
-    if dist_url is None:
```

Comment:
Can we preserve this multi-node training trick?

Reply:
Now you could find it in README